### PR TITLE
Replace ifdef with hw abstraction layer

### PR DESCRIPTION
**Pull request by pd0wm**
Retry of https://github.com/commaai/openpilot/pull/20801

### CODE REVIEW

**Comment by deanlee** [comment]
my bad.there are still problems in this PR.

**Comment by pd0wm** [comment]
Did you test in on a device? It looked like EON/TICI was never true, think inheritance behaves differently combined with static?

File: selfdrive/camerad/cameras/camera_common.cc [hunk 1]
```diff
@@ -64,13 +62,13 @@ void CameraBuf::init(cl_device_id device_id, cl_context context, CameraState *s,
 
   rgb_width = ci->frame_width;
   rgb_height = ci->frame_height;
-#ifndef QCOM2
-  // debayering does a 2x downscale
-  if (ci->bayer) {
+
+  if (Hardware::EON() && ci->bayer) {
```
Comment (pd0wm) [inline selfdrive/camerad/cameras/camera_common.cc:66]:
```suggestion
  if (!Hardware::TICI() && ci->bayer) {
```

Should be the same, but consistent with choosing the CL file that actually chooses between 2x downscale.

File: selfdrive/modeld/models/dmonitoring.cc [hunk 2]
```diff
@@ -17,12 +18,7 @@
 #endif
 
 void dmonitoring_init(DMonitoringModelState* s) {
-#if defined(QCOM) || defined(QCOM2)
-  const char* model_path = "../../models/dmonitoring_model_q.dlc";
-#else
-  const char* model_path = "../../models/dmonitoring_model.dlc";
-#endif
-
+  const char *model_path = "../../models/dmonitoring_model_q.dlc";
```
Comment (pd0wm) [inline selfdrive/modeld/models/dmonitoring.cc:21]: `dmonitoring_model.dlc` doesn't exist anymore

File: selfdrive/ui/qt/offroad/settings.cc [hunk 3]
```diff
@@ -49,13 +49,13 @@ TogglesPanel::TogglesPanel(QWidget *parent) : QWidget(parent) {
...
                                   this));
 
-#ifndef QCOM2
-  toggles.append(new ParamControl("IsUploadRawEnabled",
-                                 "Upload Raw Logs",
-                                 "Upload full logs and full resolution video by default while on WiFi. If not enabled, individual logs can be marked for upload at my.comma.ai/useradmin.",
-                                 "../assets/offroad/icon_network.png",
-                                 this));
-#endif
+  if (Hardware::TICI()) {
```
Comment (jwooning) [inline selfdrive/ui/qt/offroad/settings.cc:52]: Shouldn't this be `!Hardware::TICI())`?

File: selfdrive/common/modeldata.h [hunk 4]
```diff
@@ -20,39 +20,26 @@ const double X_IDXS[TRAJECTORY_SIZE] = { 0.    ,   0.1875,   0.75  ,   1.6875,
...
+                               0., 0., 0.,
+                               0., 0., 0.}};
 
 static inline mat3 get_model_yuv_transform(bool bayer = true) {
-#ifndef QCOM2
-  float db_s = 0.5; // debayering does a 2x downscale
-#else
-  float db_s = 1.0;
-#endif
+  float db_s = Hardware::TICI() ? 0.5 : 1.0; // debayering does a 2x downscale on TICI
```
Comment (jwooning) [inline selfdrive/common/modeldata.h:42]: other way around?

File: selfdrive/ui/qt/offroad/settings.cc (hunk 3 above)
Comment (pd0wm) [reply selfdrive/ui/qt/offroad/settings.cc:52]:
```suggestion
  if (!Hardware::TICI()) {
```

File: selfdrive/common/modeldata.h (hunk 4 above)
Comment (pd0wm) [reply selfdrive/common/modeldata.h:42]:
```suggestion
  float db_s = Hardware::TICI() ? 1.0 : 0.5; // debayering does a 2x downscale on TICI
```

**Pull request merged**